lms/djangoapps/courseware/views.py

```diff
@@ -609,14 +612,32 @@ def jump_to(request, course_id, location):
     # choose the appropriate view (and provide the necessary args) based on the
     # args provided by the redirect.
     # Rely on index to do all error handling and access control.
+
+    # this block is taken from upstream url_helpers.py - and it does not guarantee to work if anything
+    # (course_key, chapter, section, position, final_target_id) is not specified. It should be enough for prototyping
     if chapter is None:
-        return redirect('courseware', course_id=course_key.to_deprecated_string())
+        redirect_url = reverse('courseware', args=(unicode(course_key), ))
     elif section is None:
-        return redirect('courseware_chapter', course_id=course_key.to_deprecated_string(), chapter=chapter)
+        redirect_url = reverse('courseware_chapter', args=(unicode(course_key), chapter))
     elif position is None:
-        return redirect('courseware_section', course_id=course_key.to_deprecated_string(), chapter=chapter, section=section)
+        redirect_url = reverse(
+            'courseware_section',
+            args=(unicode(course_key), chapter, section)
+        )
     else:
-        return redirect('courseware_position', course_id=course_key.to_deprecated_string(), chapter=chapter, section=section, position=position)
+        # Here we use the navigation_index from the position returned from
+        # path_to_location - we can only navigate to the topmost vertical at the
+        # moment
+
+        redirect_url = reverse(
+            'courseware_position',
+            args=(unicode(course_key), chapter, section, position)
+        )
+
+    if final_target_id:
+        redirect_url += "?activate={final_target_id}".format(final_target_id=final_target_id)
```

Comment:
It could be more consistent to extend the existing URL path instead of using a parameter here - something like https://edge.edx.org/courses/College/DAP002/2014/courseware/97fd55625fe7426cabf5c9f25f95cb3c/d870e5bc011e4b13a7f9c0ea4bca3272/2/653008afe147487f8e1a91eae11d838c/ with the target id at the end.

Also, it's possible that some people would prefer to follow the pattern described in `common/lib/xmodule/xmodule/modulestore/search.py`:

         # This block of code will find the position of a module within a nested tree
         # of modules. If a problem is on tab 2 of a sequence that's on tab 3 of a
         # of modules. If a problem is on tab 2 of a sequence that's on tab 3 of a
         # sequence, the resulting position is 3_2. However, no positional modules
         # sequence, the resulting position is 3_2. 

Ie, that would give something like https://edge.edx.org/courses/College/DAP002/2014/courseware/97fd55625fe7426cabf5c9f25f95cb3c/d870e5bc011e4b13a7f9c0ea4bca3272/2_3_5_1/ . I prefer your approach with the target_id, which is easier to work with imho, but it might be worth discussing this with upstream ahead of time to see if there is already a strong preference for the approach.